src/pages/SetPasswordPage.js

```diff
@@ -62,24 +63,46 @@ class SetPasswordPage extends Component {
         this.state = {
             password: '',
             isFormValid: false,
+            error: '',
         };
     }
 
     /**
      * Validate the form and then submit it
      */
     validateAndSubmitForm() {
+        const accountID = lodashGet(this.props.route.params, 'accountID', '');
+        const validateCode = lodashGet(this.props.route.params, 'validateCode', '');
         if (!this.state.isFormValid) {
             return;
         }
-        setPassword(
-            this.state.password,
-            lodashGet(this.props.route, 'params.validateCode', ''),
-            lodashGet(this.props.route, 'params.accountID', ''),
-        );
+        API.ValidateEmail({
```

Comment:
I am pretty sure this broke the page for when the account is already validated... The API returns `405 Account already validated`.